polyply/src/dispatch_gmx.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2020 University of Groningen
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import subprocess
+import vermouth
+import os
+
+class GMXError(Exception):
+    """
+    Class for handeling GROMACS errors. This class only
+    prints the essentail part to the error message when
+    raised.
+    """
+
+    def __init__(self, stderr):
+        lines = stderr.decode("utf-8")
+        message = "GROMACS terminated with the following error:\n"
+        message = message + lines
+        super().__init__(message)
+
+
+class DispatchGMX:
+    """
+    Abstract base class for running gromacs from
+    within polyply.
+    """
+
+    def __init__(self, gmx_exe, workdir, gmx_path):
+        self.gmx_exe = gmx_exe
+        self.workdir = workdir
+        self.gmx_path = gmx_path
+
+    def run(self, command_args, stdin=None):
+        """
+        Run GMX command with parameters.
+        """
+        os.chdir(self.workdir)
+        command = [self.gmx_path, self.gmx_exe]
+
+        for token, value in command_args.items():
+            command.append("-" + token)
+            command.append(value)
+
+        output = subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, input=stdin)
```

Comment:
add: `text=True`  (Py3.7+) and the stdout/in/err streams will be strings instead of bytes.
Consider adding `cwd=self.workdir` instead of doing `os.chdir` above.